trade-finance-manager-api/src/v1/controllers/team.controller.js

```diff
@@ -1,8 +1,43 @@
-const api = require('../api');
+import { HttpStatusCode } from 'axios';
+import { USER_STATUS } from '@ukef/dtfs2-common';
+import api from '../api';
 
-const findTeamMembers = async (teamId) => {
-  const teamMembers = await api.findTeamMembers(teamId);
-  return teamMembers;
-};
+/**
+ * Maps team members to a specific format and filters out inactive users.
+ *
+ * @param {Object} req - The request object.
+ * @param {Object} req.params - The request parameters.
+ * @param {string} req.params.teamId - The ID of the team.
+ * @param {Object} res - The response object.
+ *
+ * @returns {Promise<void>} - A promise that resolves to void.
+ *
+ * @throws {Error} - Throws an error if there is an issue with mapping team members.
+ */
+export const mapTeamMembers = async (req, res) => {
+  try {
+    const { teamId } = req.params;
+    const members = await api.findTeamMembers(teamId);
+
+    // Upon failure
+    if (members?.status) {
+      const { status, data } = members;
+      return res.status(status).send({ status, data });
+    }
+
+    // Only active users
+    const activeMembers = members.filter(({ status }) => status === USER_STATUS.ACTIVE);
 
-exports.findTeamMembers = findTeamMembers;
+    // Expected response
+    const teamMembers = activeMembers.map(({ _id, firstName, lastName }) => ({
```

Comment:
Looks like we could just do a single filter with a a condition instead of having to filter and then map:

```ts
const activeMembers = members.filter(({
  status,
  _id,
  firstName,
  lastName,
}) => {
  if (status === USER_STATUS.ACTIVE) {
    return {
      _id,
      firstName,
      lastName,
    };
  }

  return null;
});
```

Reply:
Good point, however in this instance I have explicity segreated `filter` and `map` as there will be more filter conditions to add which will make it easier to read and add/remove conditions in future, for example:

```js
// Only active users
    const activeMembers = members
      .filter(({ status }) => status === USER_STATUS.ACTIVE)
      .filter(({ disabled }) => !disabled);
    
```

With the advent of SSO this will be refactored to fetch other details.

Reply:
Makes sense if more things will be added - could do the above in one filter btw e.g

```ts
.filter(({ disabled, status }) => !disabled && status === USER_STATUS.ACTIVE)
```